src/test/components/elements/panels/SelectedEventDetails/index.test.tsx

```diff
@@ -1,17 +1,57 @@
 import { screen } from "@testing-library/react";
 import {
   SelectedEventDetails,
-  countGenders,
   countStudentsAndTeachers,
 } from "../../../../../app/components/elements/panels/SelectedEventDetails";
 import { ACTION_TYPE, API_PATH, asPercentage, augmentEvent, formatAddress } from "../../../../../app/services";
 import { renderTestEnvironment } from "../../../../utils";
 import { mockEvent, mockEventBookings } from "../../../../../mocks/data";
 import { AugmentedEvent } from "../../../../../IsaacAppTypes";
-import { EventBookingDTO, IsaacEventPageDTO } from "../../../../../IsaacApiTypes";
+import { EventBookingDTO, IsaacEventPageDTO, UserRole } from "../../../../../IsaacApiTypes";
 import { FRIENDLY_DATE_AND_TIME } from "../../../../../app/components/elements/DateString";
 import { store } from "../../../../../app/state";
 import { rest } from "msw";
+import { countEventDetailsByRole } from "../../../../../app/components/elements/panels/EventGenderDetails";
+
+const formatGenderDetails = (role: UserRole, eventBookings: EventBookingDTO[]) => {
+  const { genders, numberOfConfirmedOrAttendedBookings } = countEventDetailsByRole(role, eventBookings);
+
+  return [
+    `${genders.male} (${asPercentage(genders.male, numberOfConfirmedOrAttendedBookings)}%)`,
+    `${genders.female} (${asPercentage(genders.female, numberOfConfirmedOrAttendedBookings)}%)`,
+    `${genders.other} (${asPercentage(genders.other, numberOfConfirmedOrAttendedBookings)}%)`,
+    `${genders.preferNotToSay} (${asPercentage(genders.preferNotToSay, numberOfConfirmedOrAttendedBookings)}%)`,
+    `${genders.unknown} (${asPercentage(genders.unknown, numberOfConfirmedOrAttendedBookings)}%)`,
+  ];
+};
+
+const findExpectedValues = (event: AugmentedEvent, eventBookings: EventBookingDTO[]) => {
```

Comment:
Since you're mostly just refactoring this, don't change it, but in the future please avoid including logic in the tests like you've done with `findExpectedValues` and `formatGenderDetails`, as it's prone to errors and hard to read. In this case, given that this is only used in a couple of places, it would have been much better to simply type the values that you expect. This is how you normally tend to write tests when you do test driven development, for instance.

Reply:
I've changed it :)